x-pack/plugins/canvas/public/state/actions/elements.js

```diff
@@ -192,51 +192,23 @@ export const fetchAllRenderables = createThunk(
   }
 );
 
-export const duplicateElement = createThunk(
-  'duplicateElement',
-  ({ dispatch, type }, element, pageId) => {
-    const newElement = { ...getDefaultElement(), ...getBareElement(element) };
-    // move the element so users can see that it was added
+export const insertNodes = createThunk('insertNodes', ({ dispatch, type }, elements, pageId) => {
```

Comment:
nit: This function looks awfully similar to the `addElement` function. Can we just delete `addElement` and maybe rename this `addElements` since it handles adding multiple elements?

The only difference is this function adds an offset to the position of the new element, but I feel like the offset could be added elsewhere. It'd be a nice refactor.

Reply:
Thanks @cqliu1 good point! I'll handle it as a separate small PR, as would I do with @w33ble's suggestions.